transformations/whitespace_perturbation/transformation.py

```diff
@@ -1,33 +1,37 @@
+import itertools
 import random
 
 from interfaces.SentenceOperation import SentenceOperation
 from tasks.TaskTypes import TaskType
 
 
-def whitespace(text, remove_prob=0.1, add_prob=0.05, seed=0):
+def whitespace(text, remove_prob=0.1, add_prob=0.05, seed=0, max_outputs=1):
     random.seed(seed)
-    newtext = ''
-    for char in text:
-        random_num = random.random()
-        if char.isspace() and random_num < remove_prob:
-            continue
-        newtext += char
-        if (not char.isspace()) and random_num < add_prob:
-            newtext += ' '
+    perturbed_texts = []
+    for _ in itertools.repeat(None, max_outputs):
+        perturbed_text = []
+        for char in text:
+            random_num = random.random()
+            if char.isspace() and random_num < remove_prob:
+                continue
+            perturbed_text.append(char)
+            if (not char.isspace()) and random_num < add_prob:
+                perturbed_text.append(' ')
 
-    return newtext
+        perturbed_texts.append(''.join(perturbed_text))
+    return perturbed_texts
 
 
 class WhitespacePerturbation(SentenceOperation):
     tasks = [
         TaskType.TEXT_CLASSIFICATION,
         TaskType.TEXT_TO_TEXT_GENERATION,
     ]
-    languages = ["en"]
+    languages = ["en", "es", "de", "fr", "it"]
```

Comment:
I would recommend adding more languages (as many as possible) this transformation supports.

Reply:
By this I mean "what other languages does this existing transformation apply to beyond `["en", "es", "de", "fr", "it"]`".

Reply:
This is a very good question. For now, I only put the few languages I at least know some basics and have very high confidence that could work. Unfortunately, I don’t have knowledge in a lot other languages and how they are tokenized usually, and thus could not include those confidently. Do you have any suggestions? Maybe can the organizers provide an exhaustive list of languages of interest, some examples for each language, and tokenizer(s)? If those are provided, people can easily check whether a transformation works for a language by looking at the transformed examples.

Reply:
In general, I believe languages that use the Latin (including non-European languages like Swahili, Indonesian, Malay, Vietnamese, etc) and Cyrillic scripts (and also Greek) have space-delimited words. Some Latin examples are in this PR: https://github.com/GEM-benchmark/NL-Augmenter/pull/84/files/c9f1e4bdf8fbdacce706c3a6000a42455019c535..c38dd6535fc18b107b3f3bfe924cc370a323df89

@kaustubhdhole might be able to chime in on the South Asian languages

Reply:
I think most Indian languages would be applicable here : "mr", "hi", "ur", "gj", "ta", "te", "kn", "pa", "kok"

Reply:
@samsontmr @kaustubhdhole Thanks for your inputs! I added more languages to the list.

Reply:
@kaustubhdhole I added the languages you suggested except "gj" and "kok". Are they Gujarat and Konkani? Given that they are not listed in the [ISO 639-1 codes](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes), I did not add them yet. Feel free to let me know if I could add those without causing any confusion.

Reply:
Okay - that's fine - I just noticed Gujarati is "gu".

Reply:
added "gu". thanks!